tuf/ngclient/updater.py

```diff
@@ -292,8 +293,15 @@ def _load_local_metadata(self, rolename: str) -> bytes:
             return f.read()
 
     def _persist_metadata(self, rolename: str, data: bytes):
-        with open(os.path.join(self._dir, f"{rolename}.json"), "wb") as f:
-            f.write(data)
+        """To ensure there is no chance for loss of data during writing, the data is first written to a
+        temp file followed by an atomic move operation to the correct file location to make sure that 
+        if data writing process is halted or interrupted, the original data is not lost.
+        """
+        
+        original_filename = open(f"{rolename}.json", 'a+')
+        file_out = tempfile.NamedTemporaryFile(dir=os.path.dirname(original_filename), delete=False)
```

Comment:
Hmm, don't we want to clean up after temp files after we are done moving?

Reply:
If there is any problem or the original file is somehow corrupted during any operation, the temp file will act as backup. That's the idea behind keeping the temp file intact, as I can see.